### PR TITLE
[fix](gc tablet) fix get shutdown tablet cost a lot time

### PR DESCRIPTION
**Pull request by yujun777**
If gc shutdown tablets take a lot time,   it will make TabletManager getting a  tablet wait a lot time.

pr #26151 had fix get a running tablet wait a lot time. Also it fix drop a tablet hold a  tablet meta lock too long.

But it hadn't fix get a deleted tablet wait a log time.  So this pr  fix this.

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yujun777** [comment]
run buildall

File: be/src/olap/tablet_manager.cpp [hunk 1]
```diff
@@ -1040,32 +1028,28 @@ Status TabletManager::build_all_report_tablets_info(std::map<TTabletId, TTablet>
...
-                _shutdown_deleting_tablets.push_back(tablet);
-            }
-            _shutdown_tablets.clear();
+            std::shared_lock rdlock(_shutdown_tablets_lock);
+            tablet_num = _shutdown_tablets.size();
         }
-        std::lock_guard<std::shared_mutex> wrlock(_shutdown_deleting_tablets_lock);
-        auto it = _shutdown_deleting_tablets.begin();
-        while (it != _shutdown_deleting_tablets.end()) {
+        auto it = _shutdown_tablets.begin();
```
Comment (eldenmoon) [inline be/src/olap/tablet_manager.cpp:1051]: is _shutdown_tablets thread safe to iterate?

Comment (yujun777) [reply be/src/olap/tablet_manager.cpp]: update

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

**Comment by yujun777** [comment]
run buildall

File: be/src/olap/tablet_manager.cpp [hunk 2]
```diff
@@ -1040,119 +1028,155 @@ Status TabletManager::build_all_report_tablets_info(std::map<TTabletId, TTablet>
...
-            std::lock_guard<std::shared_mutex> wrlock1(_shutdown_tablets_lock);
-            std::lock_guard<std::shared_mutex> wrlock2(_shutdown_deleting_tablets_lock);
-            for (const auto& tablet : _shutdown_tablets) {
-                _shutdown_deleting_tablets.push_back(tablet);
+    auto get_batch_tablets = [this, &last_it](int limit) {
+        std::vector<TabletSharedPtr> batch_tablets;
+        std::lock_guard<std::shared_mutex> wrdlock(_shutdown_tablets_lock);
+        while (last_it != _shutdown_tablets.end() && batch_tablets.size() < limit) {
+            // it means current tablet is referenced by other thread
+            if (last_it->use_count() > 1) {
```
Comment (eldenmoon) [inline be/src/olap/tablet_manager.cpp:1054]: what if tablet is referenced during `_move_tablet_to_trash`, will the query being correct?

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**